router/router-manager.go

```diff
@@ -0,0 +1,61 @@
+package router
+
+import "fmt"
+
+var (
+	RoutersManager RoutersManagerI
+)
+
+type RoutersManagerI interface {
+	AddRouter(router *HandleT)
+	PauseAll()
+	Pause(destType string)
+	ResumeAll()
+	Resume(destType string)
+}
+
+type RoutersManagerT struct {
+	Routers map[string]*HandleT
+}
+
+func RoutersManagerSetup() {
+	pkgLogger.Info("setting up RoutersManager.")
+	rm := new(RoutersManagerT)
+
+	RoutersManager = rm
+	rm.Routers = make(map[string]*HandleT)
+}
+
+func GetRoutersManager() (RoutersManagerI, error) {
+	if RoutersManager == nil {
+		return nil, fmt.Errorf("routersManager is not initialized. Retry after sometime")
+	}
+
+	return RoutersManager, nil
+}
+
+func (rm *RoutersManagerT) AddRouter(router *HandleT) {
+	if _, ok := rm.Routers[router.destName]; !ok {
```

Comment:
Would there be a place, where the router instance for the same destination changes?

To be on safe side, should we keep the latest one instead of the oldest.

Reply:
no.

Anyways, changed to use latest